aaelf64-morello/aaelf64-morello.rst

```diff
@@ -0,0 +1,831 @@
+
+
+Static Morello relocations
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. warning:: The ELF64 Code of the relocations are subject to change.
+
+.. class:: aaelf64-morello-static-relocations
+
+.. table:: Relocations to generate 19, 21, and 33 bit PC-relative addresses
+
+    +-------+-----------------------------------+-----------------+-------------------------------------------------------------------------+
+    | ELF64 | Name                              | Operation       | Comment                                                                 |
+    | Code  |                                   |                 |                                                                         |
+    +=======+===================================+=================+=========================================================================+
+    | 57348 | ``R_MORELLO_LD_PREL_LO17``        | ``S+A -``       | Set a load-literal immediate value to bits [20:4] of X;                 |
+    |       |                                   | ``(P&~0xF)``    | check that -2\ :sup:`20` <= X < 2\ :sup:`20`, check that ``X & 15 = 0`` |
+    +-------+-----------------------------------+-----------------+-------------------------------------------------------------------------+
+    | 57349 | ``R_MORELLO_ADR_PREL_PG_HI20``    | ``Page(S+A)``   | Set an ADRP immediate value to bits [31:12] of the X;                   |
+    |       |                                   | ``- Page(P)``   | check that -2\ :sup:`31` <= X < 2\ :sup:`31`                            |
+    +-------+-----------------------------------+-----------------+-------------------------------------------------------------------------+
+    | 57350 | ``R_MORELLO_ADR_PREL_PG_HI20_NC`` | ``Page(S+A)``   | Set an ADRP immediate value to bits [31:12] of the X;                   |
+    |       |                                   | ``- Page(P)``   | No overflow check                                                       |
+    |       |                                   |                 | Although overflow must not be checked, a linker should check that the   |
+    |       |                                   |                 | value of X is aligned to a multiple of the datum size.                  |
+    +-------+-----------------------------------+-----------------+-------------------------------------------------------------------------+
+
+.. class:: aaelf64-morello-control-flow-relocations
+
+.. table:: Relocations for control-flow instructions - all offsets are a multiple of 4
+
+    +-------+-------------------------+------------------+-------------------------------------------------+
+    | ELF64 | Name                    | Operation        | Comment                                         |
+    | Code  |                         |                  |                                                 |
+    +=======+=========================+==================+=================================================+
```

Comment:
Please end sentences starting with a Capital with a period`.` This also makes two sentences in a row hard to read like what happens with `See..`. I'm not sure if you meant to have two senteces separated by a return in the table follow eachother directly without a return, but that's how rst interprets it.

Reply:
Done. But does not show as `outdated` in the UI for some reason.